step-api/src/main/java/org/jenkinsci/plugins/workflow/structs/DescribableHelper.java

```diff
@@ -272,7 +272,21 @@ public String getDisplayName() {
      * A type of a parameter to a class.
      */
     public static abstract class ParameterType {
-        ParameterType() {}
+        @Nonnull
+        private final Type actualType;
+
+        public Type getActualType() {
+            return actualType;
+        }
+
+        public Class<?> getType() {
+            return (Class) getActualType();
```

Comment:
I would expect `actualType` to be the `type` passed to `of`, which is a `Class` except in the second (collection-oriented) call to the `ArrayType` constructor, when it is a `ParameterizedType`.